From 8e5d184b4b0ec7660cd10b02fac6edd9bb1a4df5 Mon Sep 17 00:00:00 2001
From: Guruswamy Basavaiah <guruswamy.basavaiah@broadcom.com>
Date: Thu, 10 Apr 2025 10:36:39 +0000
Subject: [PATCH] Support multiline MODULE_DESCRIPTION

Support for multiline strings in the MODULE_DESCRIPTION
macro within kernel livepatch modules.
Signed-off-by: Guruswamy Basavaiah <guruswamy.basavaiah@broadcom.com>
---
 kpatch-build/kpatch-build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index 65506b3..3a981b5 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -1436,7 +1436,7 @@ CROSS_COMPILE="$CROSS_COMPILE"
 save_env
 
 if [[ $DESC_FILE ]]; then
-	echo "MODULE_DESCRIPTION(\"$(cat $DESC_FILE)\");" >> $TEMPDIR/patch/livepatch-patch-hook.c || die
+	echo "MODULE_DESCRIPTION($(sed 's/^/"/; s/$/"/' $DESC_FILE));" >> $TEMPDIR/patch/livepatch-patch-hook.c || die
 fi
 
 
-- 
2.39.4

